staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -369,8 +369,9 @@ func secretboxPrefixTransformer(config *apiserverconfig.SecretboxConfiguration)
 func envelopePrefixTransformer(config *apiserverconfig.KMSConfiguration, envelopeService envelope.Service, prefix string) (value.PrefixTransformer, error) {
 	baseTransformerFunc := func(block cipher.Block) value.Transformer {
 		// v1.24: write using AES-CBC only but support reads via AES-CBC and AES-GCM (so we can move to AES-GCM)
-		// TODO(aramase): swap this ordering in v1.25
-		return unionTransformers{aestransformer.NewCBCTransformer(block), aestransformer.NewGCMTransformer(block)}
+		// v1.25: write using AES-GCM only but support reads via AES-GCM and fallback to AES-CBC for backwards compatibility
+		// TODO(aramase): v1.26+: drop support for AES-CBC
```

Comment:
We cannot drop CBC read support until we automate storage migration.

We could have a release note that hard require that storage migration be performed, but seems a bit user hostile to me.

Reply:
Noted! Added this as a comment.